Update manage-your-attributes.md (#335)
akeneo/pim-helpcenter: content/md/first-steps/manage-your-attributes.md

# How to create an attribute?

Akeneo proposes several types of attributes. Discover all attributes available on the article [Introduction to Akeneo attribute types](what-is-an-attribute.html#definition-of-an-attribute). Make sure you pick the **most appropriate attribute type to perfectly match your needs**.    
For example, if you need to make sure that your PIM users will select one of a predefined option, select a simple select attribute type instead of a free text field. Create a `metric` attribute to fill in the weight/size/length... of your products.

To create an attribute in Akeneo PIM:
1. Go to `Settings/Attributes`
1. Click on the `Create attribute` button:
  ![image](../img/Settings_AttributesCTAButton.png)
1. Select the attribute type you want:
  ![image](../img/Settings_Attributes_Create2.png)
1. The attribute edit form is displayed, fill in the following properties:  
    Under `General Parameters`: 
    - A **code** (required) to identify the attribute.
    - A **type** based on the attribute type selected in the previous screen.
    - An **attribute group** (required) in which the new attribute will be added to.

>     In addition to these required attributes, you can select the following properties:

Reviewer: Il manque un retour à la ligne

>     In addition to these required attributes, you can select the following properties:

Author: ```suggestion
In addition to these required attributes, you can select the following properties:
```

    - **Unique value** (Yes/No): enable this property if the value of this attribute must be unique. By default, it is set to yes for `Identifier` attribute type.
    - **Value per channel** (Yes/No): enable this property if you plan to have different product values per channel.
    - **Value per locale** (Yes/No): enable this property if you plan to have different product values per locale.

>     - **Read only** (Yes/No): to prevent this attribute from being edited through the UI, it will only be updated through imports, API or rules _(Enterprise Edition only)_

Author: ```suggestion
    - **Read only** (Yes/No) **_(Enterprise Edition only)_**: to prevent this attribute from being edited through the UI, it will only be updated through imports, API or rules
```

    - **Usable in grid** (Yes/No): if you want to filter on this attribute in the product's grids or use it as a column.
    - **Locale specific**: this attribute will only be available for some locales.      

> Under `Type Specific Parameters`, add validations to control your data: 

Reviewer: Le nom de la section est "Validation Parameters" et y a un pb d'indentation

Author: ```suggestion
Under `Validation Parameters`, add validations to control your data: 
```

    - **Validation parameters**: each attribute comes with validation parameters (see [Add attributes validation parameters](#parameters-according-to-specific-attribute-types)).    
1. Click on `Save` to create your attribute.

> 

Reviewer: Tu as supprimé le tip ? Il était pas mal car c'est une errur courante sur les select.

:::warning

> **Some attribute properties cannot be edited afterwards, such as the `code`, the `unique value`, `value per channel` and `value per locale`** make sure they fit your needs!

Author: ```suggestion
**Some attribute properties cannot be edited afterwards, such as `code`, `unique value`, `value per channel` and `value per locale`** make sure they fit your needs!
```

  - *Validation rule*: how the value should be validated: such as an URL, as an Email, or using a regular expression
- **Text area attribute**:
  - *Maximum number of characters*: maximum number of characters that an attribute value can have
  - *Enable WYSIWYG*: activates the WYSIWYG feature
- **File attribute** and **image attribute**:
  - *Maximum size allowed in MB*
  - *Allowed extensions files*
- **Metric attribute** :
  - *Allow negative values* for the attribute
  - *Allow decimal values* for this attribute
  - *Minimum value*: the smallest value allowed
  - *Maximum value*: the largest value allowed
  - *Metric family*: family measurement unit to be used for the attribute (weight, dimensions, area, etc. )
  - The *default unit of measurement*
- **Price attribute**:
  - *Allow decimals* for the attribute values
  - *minimum value*: the smallest allowed value
  - *maximum value*: the largest allowed value
- **Number attribute**:
  - *Allow negative values* for the attribute
  - *Allow decimals*
  - *Minimum value*: the smallest allowed value
  - *Maximum value*: the largest allowed value

> - **Reference entity single link attribute** & **reference entity multiple links attribute**_(EE only)_:

Author: ```suggestion
- **Reference entity single link attribute** & **reference entity multiple links attribute _(Enterprise Edition only)_**:
```

  - *Related reference entity*: the reference entity linked to this attribute

> - **Reference data simple select attribute** & **reference data multi select **:

Author: ```suggestion
- **Reference data simple select attribute** & **reference data multi select**:
```

  - *Reference data type*: choose an existing reference data to link to your attribute.
  
For the text attribute type, you can add a regular expression to validate the content of the input. For instance, to control the right format of a EAN13 code, add a validation rule based on the following regular expression: `/^[0-9]{13}$/`

![Validation parameters](../img/Settings_EAN.png)

::: info
The **simple** and **multi select** attribute types, as well as **Yes/No** (boolean) attribute have no specific validation properties.
:::

## Translate your attribute labels

> Once your attribute is created, translate it! you can add a **translation for each enabled locales** under the `Label translations` tab. If you need to edit some other attribute's property, please refer to the section [Edit attribute properties](manage-your-attributes.html#edit-attribute-properties)

Reviewer: ```suggestion
Once your attribute is created, translate it! You can add a **translation for each enabled locales** under the `Label translations` tab. If you need to edit some other attribute's property, please refer to the section [Edit attribute properties](manage-your-attributes.html#edit-attribute-properties).
```

- **Validation parameters** (see [Parameters according to specific attributes types](#parameters-according-to-specific-attribute-types))
- **Options** (for simple or multi select attribute types only): these are predefined values that the user can select

# Find your attributes

Go to `Settings/Attributes`, to see all the existing attributes or create a new one.

The attribute grid displays 25 attributes per page. To browse attributes pages, click on the page number you want to reach or use the filters.

![image](../img/Settings_Attributes.png)    

You can sort out attributes by clicking on the column header on the grid.

## Search on the grid

### On its label

To quickly find an attribute, use the search bar available above the grid. Use your attributes label to find what you are looking for.

![image](../img/Settings-UsetheSearchBar.gif)

::: warning
If your attribute does not have any label translated, it will not pop in the search results. **Make sure your attribute has its labels translated.**
:::

### Using the filters

Other filters are available on the right side of the screen, you can search attributes based on their code, type...

> ![image](../img/Settings_AttributesFilters.png)

Reviewer: Ca pourrait être pas mal de changer la copie d'écran vu qu'on a le filtre sur le code en plus

Author: je sais plus faire les copies d'écran :D

Reviewer: la prochaine fois alors ;)

Reviewer: Je peux te montrer

An option that is used by a published product cannot be deleted. Unpublish the product to be able to delete the option.
:::

## Sort options

> Options can be **automatically sorted by alphabetical order** using the toggle button `Sort automatically options by alphabetical order` on the top of the attribute's options page.

Reviewer: ```suggestion
Options can be **automatically sorted by alphabetical order** using the toggle button `Sort automatically options by alphabetical order` on the top of the attribute's options page.  
```

> If you want to define your own sort, you **can drag and drop option rows to reorder options manually**.

Reviewer: Il te manque un saut de ligne
